src/Container.ts

```diff
@@ -40,38 +49,42 @@ export class Container {
     /**
      * Registers a new service.
      *
-     * @param name Service name. Optional
-     * @param type Service class
-     * @param params Parameters to be sent to constructor on service initialization
+     * @param descriptor
      */
-    static registerService(name: string, type: Function, params?: any[]) {
-        this.registeredServices.push({ name: name, type: type, params: params });
+    static registerService(descriptor: ServiceDescriptor) {
+        this.registeredServices.push(descriptor);
     }
 
     /**
      * Retrieves the service with the specific name or given type from the service container.
-     * Optionally parameters can be pass in the case if instance is initialized in the container for the first time.
+     * Optionally, parameters can be passed in case if instance is initialized in the container for the first time.
      */
     static get<T>(type: ConstructorFunction<T>, params?: any[]): T;
     static get<T>(name: string, params?: any[]): T;
     static get<T>(typeOrName: ConstructorFunction<T>|string, params?: any[]): T {
 
         // normalize parameters
-        let type: Function, name: string;
+        let type: Function;
+        let name: string;
+        let factory: Function;
         if (typeof typeOrName === "string") {
             name = <string> typeOrName;
         } else {
             type = <Function> typeOrName;
         }
 
         // find if service was already registered
-        const registeredService = this.findRegisteredService(name, type);
-        // console.log("registeredService: ", registeredService);
-        if (registeredService) {
-            if (!type)
-                type = registeredService.type;
-            if (!params)
-                params = registeredService.params;
+        const serviceDescriptor = this.findRegisteredService(name, type);
+        if (serviceDescriptor) {
+            if (!type) {
+                type = serviceDescriptor.type;
+            }
+            if (!params) {
+                params = serviceDescriptor.params;
+            }
+            if (serviceDescriptor.factory) {
+                factory = serviceDescriptor.factory;
+            }
```

Comment:
Actually it makes the code more standardized (you have only one way to define code block instead of two) and it makes it easier to modify, because you can quickly add something to the existing code block: for example you can add an ugly `console.log()` statement to debug something. Otherwise you will have to first convert single line to a code block and then add a statement to it. And if that statement was temporary you will have to convert it back to a single line after deleting it. This is a lot of extraneous work. I was using single-line approach in PHP earlier in my career, but then switched to always using code blocks and I very rarely see code where single line statements are used instead of code blocks nowadays. And it's not making code harder to read (objectively), it's just a matter of preference and habit, I think.